docs/DeveloperGuide.md

```diff
@@ -257,33 +257,147 @@ _{Explain here how the data archiving feature will be implemented}_
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix: Requirements**
+
+### Product scope
+
+**Target user profile**:
+
+* middle-aged property agent selling new condos to prospective clients
+* has a need to manage a significant number of property listings and clients
+* prefer desktop apps over other types
+* can type fast
+* prefers typing to mouse interactions
+* is reasonably comfortable using CLI apps
+
+**Value proposition**: manage property listings and clients faster than a typical mouse/GUI driven app
+
+
+### User stories
+
+Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unlikely to have) - `*`
+
+| Priority | As a …​                                            | I want to …​               | So that I can…​                                                              |
+| -------- |----------------------------------------------------|----------------------------|------------------------------------------------------------------------------|
+| `* * *`  | new user                                           | see usage instructions     | refer to instructions when I forget how to use the App                       |
+| `* * *`  | property agent                                     | add a new property listing |                                                                              |
+| `* * *`  | property agent                                     | delete a property listing  | remove entries that I no longer need                                         |
+| `* * *`  | property agent                                     | find a listing by name     | locate details of listings without having to go through the entire list      |
+| `* *`    | property agent                                     | edit listing details       | update details of listings                                                   |
+| `*`      | property agent with many listings in the directory | sort listings by price cap | locate a listing within budget without haveing to go through the entire list |
+
+*{More to be added}*
+
+### Use cases
+
+(For all use cases below, the **System** is the `Condonery` and the **Actor** is the `user`, unless specified otherwise)
+
+**Use case: Delete a listing**
+
+**MSS**
+
+1.  User requests to list properties
+2.  Condonery shows a list of properties
+3.  User requests to delete a specific property in the list
+4.  Condonery deletes the person
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. Condonery shows an error message.
+
+      Use case resumes at step 2.
+
+*{More to be added}*
+
+### Non-Functional Requirements
+
+1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
```

Comment:
```suggestion
2.  Should be able to hold up to 1000 listings and contacts without a noticeable sluggishness in performance for typical usage.
```

Reply:
we can add this change under a different pull request for NFRs